modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTask.java

```diff
@@ -117,9 +118,10 @@ BulkByScrollTask getChildTask() {
     private void execute(ReindexJob reindexJob) {
         reindexIndexClient.getReindexTaskDoc(getPersistentTaskId(), new ActionListener<>() {
             @Override
-            public void onResponse(ReindexTaskIndexState reindexTaskIndexState) {
-                ReindexRequest reindexRequest = reindexTaskIndexState.getReindexRequest();
-                Runnable performReindex = () -> performReindex(reindexJob, reindexTaskIndexState);
+            public void onResponse(ReindexTaskState taskState) {
+                progressState = new ProgressState(taskState);
+                Runnable performReindex = () -> performReindex(reindexJob, taskState.getStateDoc());
```

Comment:
If we pass in `progressState` to `performReindex`, it looks like we can avoid the volatile `progressState` field? Can then be sent along to `handleDone` and `handleError` from `performReindex`.